### DISCUSSION TITLE
How to use it at all?

### DISCUSSION THREAD

**Discussion by lsascha** [solokeys/solo2-cli#56] [Q&A]
Sorry to ask this but i am a bit lost, also because of lack of documentation.
Only stuff i can find is a little bit about the solo1.

I have some rudimentary questions.

1. Question
As far as i understand, i can create and save authentications with the solo2-cli tool.
Questions is here:
How do i define which button of the Solo2 is doing what?
So far i was able to create an OATH HOTP with the app, but i have no idea how to define it to any button.

Also how to define what the NFC will do?

2. Question
Is it possible that it will, once i got it to work on press of a button to enter the auth parts as if it is a keyboard? Does this need something additional for me to do?

So far the only thing i got the Solo2 to do is opening the solo website when using NFC on my phone.

Either i am just too stupid to use the tool, i am too stupid to find the needed documentation or there is something else i am missing.
I would really appreciate some more documentation to help newbies like me.

**Comment by nickray** [comment]
1. You don't need a CLI for FIDO2, just use your browser. The touch "buttons" are interchangeable, touch any of them to "confirm user presence".
2. There is no keyboard functionality.

Aware documentation is lacking, but (apart from bugs to be fixed) you should only need/expect:
- use FIDO stuff via browsers or phones etc. "out of the box" (no setup/config)
- run `solo2 update` when you explicitly want to update the firmware

**Reply by lsascha** [reply]
So if i understand it correctly, all "buttons" are basicly useless and all do the same?

Thats a real bummer.
Coming from Yubikey they had only 1 button, but it could be configured to do 2 different things.

Short-Press = Do FIDO Stuff (2f. auth for websites etc.)
Long-Press = Do OATH HOTP (For example to use with Keepass)

Configure NFC to do the same as what Long-Press does.

Since Solo2 was advertised with multiple buttons, it was my expectation that all buttons had a purpose.

I guess i was wrong and i would then need seperate Solo-Keys to do each thing.
